src/session.c

```diff
@@ -68,29 +72,44 @@ static void sess_cleanup(int flags) {
   /* These are necessary in order that cleanups associated with these pools
    * (and their subpools) are properly run.
    */
-  if (session.d) {
+  if (session.d != NULL) {
     pr_inet_close(session.pool, session.d);
     session.d = NULL;
   }
 
-  if (session.c) {
+  if (session.c != NULL) {
     pr_inet_close(session.pool, session.c);
     session.c = NULL;
   }
 
   /* Run all the exit handlers */
   pr_event_generate("core.exit", NULL);
 
-  if (!is_master ||
-      (ServerType == SERVER_INETD &&
-      !(flags & PR_SESS_END_FL_SYNTAX_CHECK))) {
+  if (is_master == FALSE) {
+    log_sess_closed = TRUE;
+  }
+
+  if (ServerType == SERVER_INETD &&
+      !(flags & PR_SESS_END_FL_SYNTAX_CHECK)) {
+    log_sess_closed = TRUE;
+  }
+
+  if (sess_connected == FALSE) {
+    log_sess_closed = FALSE;
```

Comment:
Here, we skip logging the "session closed" message, if the session is not considered successfully connected.